enhancements/update/eus-upgrades-mvp.md

```diff
@@ -0,0 +1,368 @@
+to scheduling rules. This has been measured and shown that approximately 70% of
+all reschedulable pods are rescheduled more than necessary if they were
+rescheduled in an optimal manner. Optimal rescheduling should be closer to (1 /
+node_count_per_failure_domain) percentage of pods being rescheduled more than
+once during a rolling restart of a given failure domain.
+
+We should make the MCO topology aware so that it upgrades all of the host in one
+failure domain because anti-affinity scheduling is used in most scenarios. We
+should provide some mechanism to rank upgraded nodes as slightly higher priority
+in scheduler configuration. Perhaps a timestamp of configuration generation that
+the scheduler then biases toward nodes with the greatest timestamps as a last in
+chain priority.
+
+We should consider enabling the rescheduler as well to ensure that when the
+upgrade completes we rebalance so that the last node ends up with proportionate
+workload.
+
+#### ??? - Provide a hosted graph viewer which will assist disconnected admins in mirroring all content between 4.6 and 4.10
+
+Assume that 4.6.eus-z -> 4.7.z -> 4.8.z -> 4.9.z -> 4.10.z are all in the eus-4.10
+channel providing a chain of upgrades that eventually lead to 4.10. We should build
+a graph viewer / solver somewhere which allows an admin of a 4.6 cluster hoping
+to upgrade to 4.10 to discover all content which must be mirrored between their
+current version and their eventual 4.10 target.
+
+In general the graph viewer should provide a complete path from the cluster's
+current version to the latest .z in the selected channel assuming such a path exists.
+If someone were to provide cluster version 4.4.3 but select the stable-4.6 channel
+no such path exists and it should indicate that, if they were to select stable-4.5
+it would provide the upgrade path of 4.4.3 to 4.4.29 to 4.5.24 which is the latest 4.5.z.
+
+### Implementation Details/Notes/Constraints [optional]
+
+The main caveat of what's outlined here is that it does not reduce reboots
+which, for many reasons, is highly desired by some admins. The rationale for not
+addressing that in this enhancement is that this enhancement targets
+improvements which benefit all upgrades.
+
+The epic / user story to backport API removal notifications to 4.6 assumes that
+we will require anyone upgrading along an EUS specific upgrade path to update to
+a 4.6 version which is specific to EUS. This is necessary because only those who
+follow the EUS upgrade path have 4.10 as their next logical target version. We
+should not alert non EUS clusters of removals in 4.10 until they've upgraded to
+a later minor version.
+
+As with all upgrades, new features enabled by interim versions are not expected
+enabled during the upgrade path. This means that for instance new MachineConfig
+features enabling complex root devices while introduced in 4.7 may not be utilized
+until the cluster has completed its upgrade.
```

Comment:
This won't be possible, at least on the upstream side. If the admin is quick enough to use new APIs, he will succeed. Clarify what you mean with "features" here.

Reply:
My intent was to ensure that if we introduce a new capability in 4.9 which depends on updated host bound components (kernel, crio, kubelet, etc) and we forego updating RHCOS to 4.9 then that new capability is unavailable until host components are updated to 4.10.

Reply:
I'm going to remove this, it's specifically tied to any efforts to remove host level updates between releases and this enhancement makes no attempt to deliver on that. I'll add this to an enhancement that's focused on that.